src/router/middlewares/setTenantId.ts

```diff
@@ -11,6 +11,18 @@ import RouteHelper from '../routes/routeHelper';
 
 const tenantIdRegex = /^[a-zA-Z0-9\-_]{1,64}$/;
 
+const getTenantIdFromAudClaim = (audClaim: string) => {
+    // aud claim should ALWAYS be present regardless of tenancy mode
+    if (!audClaim) {
+        throw new UnauthorizedError('Unauthorized');
+    }
+    const audClaimSplit = audClaim.split('://')[1].split('/');
+    if (audClaimSplit.length === 2) {
+        return undefined;
+    }
+    return audClaimSplit.pop();
+};
```

Comment:
Looks like this string parsing is incorrect. It allows invalid aud values like: 
- `https://xxxx.execute-api.us-east-2.amazonaws.com/dev/t2`
- `https://xxxx.execute-api.us-east-2.amazonaws.com/dev/many/other/things`

We need to match **exactly** `<baseUrl>/tenant/<tenantId>` (note that the above example is missing `/tenant`)

We should use the value of `res.locals.serverUrl` as `baseUrl` to do the string matching.

https://github.com/awslabs/fhir-works-on-aws-routing/blob/f29c54e9956db73cca22b68809e4580efb320d33/src/router/middlewares/setServerUrl.ts#L21